### ISSUE TITLE
_PyEval_RequestCodeExtraIndex should return a globally valid index, not a ThreadState specific one

### ISSUE THREAD

**Issue by e896bb44-597f-4ece-9aee-2a9abddfd818** [python/cpython#72707]
BPO | [28521](https://bugs.python.org/issue28521)
--- | :---
Nosy | @jeethu

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2016-10-24.15:32:25.929>
labels = ['interpreter-core', 'type-bug']
title = '_PyEval_RequestCodeExtraIndex should return a globally valid index, not a ThreadState specific one'
updated_at = <Date 2018-01-17.02:33:28.363>
user = 'https://bugs.python.org/jpe'
```

bugs.python.org fields:
```python
activity = <Date 2018-01-17.02:33:28.363>
actor = 'jeethu'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Interpreter Core']
creation = <Date 2016-10-24.15:32:25.929>
creator = 'jpe'
dependencies = []
files = []
hgrepos = []
issue_num = 28521
keywords = []
message_count = 1.0
messages = ['279314']
nosy_count = 2.0
nosy_names = ['jpe', 'jeethu']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue28521'
versions = ['Python 3.6']
```

</p></details>

**Comment by kumaraditya303** [comment]
`_PyEval_RequestCodeExtraIndex` now uses per interpreter counter so this seems fixed. @jpe

**Comment by jpe** [comment]
Just FYI in case someone is investigating old interpreter versions, this was a bug in 3.6.0 and was fixed by 3.6.15. It was fixed in 3.7.0
